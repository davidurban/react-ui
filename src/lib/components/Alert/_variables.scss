@import '../../shared/settings/colors';
@import '../../shared/settings/offsets';
@import '../../shared/settings/typography';

$alert-line-height: 1.2;
$alert-min-height: 30px;
$alert-offset: map-get($offset-values, 1);
$alert-stripe-width: 5px;
$alert-message-font-weight: $typography-font-weight-light;
$alert-title-font-weight: $typography-font-weight-bold;
