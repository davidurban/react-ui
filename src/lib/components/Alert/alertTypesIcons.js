export default {
  error: 'highlight_off',
  help: 'help',
  info: 'info',
  note: 'note',
  success: 'check_circle',
  warning: 'warning',
};
