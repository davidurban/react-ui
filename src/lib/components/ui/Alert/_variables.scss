@import '../../../styles/settings/colors';
@import '../../../styles/settings/offsets';
@import '../../../styles/settings/typography';

$alert-line-height: 1.2;
$alert-min-height: 30px;
$alert-offset: map-get($offset-values, 1);
$alert-stripe-width: 5px;
$alert-message-font-weight: $typography-font-weight-light;
$alert-title-font-weight: $typography-font-weight-bold;
