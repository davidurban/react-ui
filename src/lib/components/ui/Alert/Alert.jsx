import PropTypes from 'prop-types';
import React from 'react';
import Icon from '../Icon';
import styles from './Alert.scss';
import alertTypesIcons from './alertTypesIcons';

const Alert = (props) => {
  let rootTypeClass = styles.note;
  let icon = alertTypesIcons.note;

  if (props.type) {
    if (props.type === 'success') {
      rootTypeClass = styles.success;
      icon = alertTypesIcons.success;
    } else if (props.type === 'error') {
      rootTypeClass = styles.error;
      icon = alertTypesIcons.error;
    } else if (props.type === 'warning') {
      rootTypeClass = styles.warning;
      icon = alertTypesIcons.warning;
    } else if (props.type === 'info') {
      rootTypeClass = styles.info;
      icon = alertTypesIcons.info;
    } else if (props.type === 'help') {
      rootTypeClass = styles.help;
      icon = alertTypesIcons.help;
    }
  }

  return (
    <div
      className={(`
        ${styles.root}
        ${rootTypeClass}
      `).trim()}
      id={props.id}
      role="alert"
    >
      <div className={styles.icon}>
        <Icon icon={icon} />
      </div>
      <div
        className={styles.message}
        {...(props.id && { id: `${props.id}__content` })}
      >
        {props.children}
      </div>
    </div>
  );
};

Alert.defaultProps = {
  id: undefined,
  type: 'note',
};

Alert.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  id: PropTypes.string,
  type: PropTypes.oneOf(Object.keys(alertTypesIcons)),
};

export default Alert;
