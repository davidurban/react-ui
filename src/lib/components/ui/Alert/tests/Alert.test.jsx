import React from 'react';
import {
  shallow,
} from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Alert from '../Alert';

jest.mock('../../Icon/load-material-design-icons');

describe('rendering', () => {
  it('renders correctly mandatory props only', () => {
    const tree = shallow(
      <Alert
        message="Message"
      >
        <div>Children</div>
      </Alert>,
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('renders correctly type', () => {
    const tree = shallow(
      <Alert
        message="Message"
        type="success"
      >
        <div>Children</div>
      </Alert>,
    );

    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
