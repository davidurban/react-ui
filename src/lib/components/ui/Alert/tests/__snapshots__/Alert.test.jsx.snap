// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly mandatory props only 1`] = `
<div
  className="root
        note"
  role="alert"
>
  <div
    className="icon"
  >
    <Icon
      icon="note"
      size="medium"
    />
  </div>
  <div
    className="message"
  />
</div>
`;

exports[`rendering renders correctly type 1`] = `
<div
  className="root
        success"
  role="alert"
>
  <div
    className="icon"
  >
    <Icon
      icon="check_circle"
      size="medium"
    />
  </div>
  <div
    className="message"
  />
</div>
`;
